tests/robustness/failpoints.go

```diff
@@ -85,30 +88,87 @@ var (
 	}}
 )
 
-func triggerFailpoints(ctx context.Context, t *testing.T, lg *zap.Logger, clus *e2e.EtcdProcessCluster, config FailpointConfig) {
+func verifyClusterHealth(ctx context.Context, t *testing.T, clus *e2e.EtcdProcessCluster, failpointName string) error {
+	for i := 0; i < len(clus.Procs); i++ {
+		clusterClient, err := clientv3.New(clientv3.Config{
+			Endpoints:            clus.Procs[i].EndpointsGRPC(),
+			Logger:               zap.NewNop(),
+			DialKeepAliveTime:    1 * time.Millisecond,
+			DialKeepAliveTimeout: 5 * time.Millisecond,
+		})
+		defer clusterClient.Close()
+
+		// Attempt to check status for the member
+		statusResp, err := clusterClient.Status(ctx, clusterClient.Endpoints()[0])
+		if err != nil {
+			return fmt.Errorf("failed to get cluster member status, err: %v", err)
+		}
+		for _, statusError := range statusResp.Errors {
+			return fmt.Errorf("cluster member status contains error, err: %v", statusError)
+		}
+
+		// Also check member /health endpoint
+		httpClient := &http.Client{}
+		healthResp, err := httpClient.Get(fmt.Sprintf("%s/health", clusterClient.Endpoints()[0]))
```

Comment:
For checking health I meant it via grpc endpoint like https://github.com/etcd-io/etcd/blob/da4bf0f76fb708e0b57763edb46ba523447b9510/tests/framework/integration/integration.go#L235-L245

Reply:
My misunderstanding, fixed 🤦🏻‍♂️